src/common/processing_linux.c

```diff
@@ -126,7 +126,7 @@ void ffProcessGetInfoLinux(pid_t pid, FFstrbuf* processName, FFstrbuf* exe, cons
 
     #ifdef __linux__
 
-    char filePath[64];
```

Comment:
You won't save memory by shrinking stack variables. There is no difference as long as the stack doesn't overflow.

Stack is statically allocated when the thread is created. It's always there, no matter if you use it or not.

Reply:
To be honest I'm not a fun of calculating *how many bytes a number string will take*. What if you make a mistake? A reasonable safe number is better for me.

Reply:
`char procFilePath[PATH_MAX]` is an obvious waste. I will keep this change and discard others.

Reply:
> Stack is statically allocated when the thread is created. It's always there, no matter if you use it or not.

The real world thing is that linux kernel only allocates a small fixed size of stack and grows it when needed (the main thread, other threads such as are created by pthread will get a fixed stack).
![image](https://github.com/user-attachments/assets/640a5ac3-7f5c-44ce-901e-a79a25c87408)

But I agree that save 100 bytes does not make sense here.

Reply:
> To be honest I'm not a fun of calculating _how many bytes a number string will take_. What if you make a mistake? A reasonable safe number is better for me.

128 on SunOS is safe and you have already used it, it's better than PATH_MAX. I wont change other buf size.

Reply:
Done.

Reply:
Really? That is interesting. AFAIK a stack memory must be continuous, which means that you must move all used memory to the new memory space without modifying stack pointers (rbp rsp etc). I wonder how it can be done.

Reply:
> Really? That is interesting. AFAIK a stack memory must be continuous, which means that you must move all used memory to the new memory space without modifying stack pointers (rbp rsp etc). I wonder how it can be done.

you can see this one: https://stackoverflow.com/questions/49595006/is-stack-memory-contiguous-physically-in-linux
Stack memory is not necessarily contiguous in the physical address space on Linux.

This one says Windows even allows discontiguous virtual memory: https://stackoverflow.com/questions/5086577/is-stack-memory-contiguous, well I only have a poor knownledge of Windows, so I cannot say this is true or not.

Reply:
> Really? That is interesting. AFAIK a stack memory must be continuous, which means that you must move all used memory to the new memory space without modifying stack pointers (rbp rsp etc). I wonder how it can be done.

By the way, stack address, frame pointer and other things are usually virtual memory, keep it be continuous is easy because the kernel can modify the page table and make discontiguous physical memory looks like continuous. And this is no so much harm for the cache.

Reply:
I dont think so. It is easy ONLY if you have control to the physical memory address. I dont think it is possible for a userland program.

Reply:
If it is possible, why do we have to move memories when appending a string or array?

Reply:
> I dont think so. It is easy ONLY if you have control to the physical memory address. I dont think it is possible for a userland program.

Stack is not controled by user space, when you add the stack pointer and access the stack memory, if it did not allocated a physical memory area, a page fault will be caused. The kernel will handle the PF and allocate physical memory, finally add it to the page table.
For detail you can see this: https://stackoverflow.com/questions/46790666/how-is-stack-memory-allocated-when-using-push-or-sub-x86-instructions

Reply:
> If it is possible, why do we have to move memories when appending a string or array?

Why we should move these memory, because we can not control the virtual memory address on the user space. The Linux kernel only handles few special cases for us. A normal mmap used by malloc wont care about the virtual memory address. Even in the kernel only a few parts can access to physical memory and arbitrarily set the virtual memory address.
 (brk/sbrk can make the small heap allocations be continuous, but they are syscalls, not on the user land).